packages/components/src/Combobox/components/ComboboxContent/ComboboxContentHeader/ComboboxContentHeader.test.tsx

```diff
@@ -0,0 +1,214 @@
+            selectedCount={0}
+            searchValue="Taylor"
+            options={[
+              { id: "1", label: "Bilbo Baggins" },
+              { id: "2", label: "Frodo Baggins" },
+              { id: "3", label: "Samwise Gamgee" },
+            ]}
+          />,
+        );
+
+        expect(queryByText("Select all")).not.toBeInTheDocument();
+        expect(queryByText("Clear")).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When one or more options has been selected", () => {
+    it("should render a button that is labelled 'Clear'", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={1}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+          ]}
+        />,
+      );
+
+      expect(getByText("Clear")).toBeInTheDocument();
+    });
+
+    it("should render a label '{selectedCount} selected'", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={3}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+            { id: "3", label: "Samwise Gamgee" },
+            { id: "4", label: "Meriadoc Brandybuck" },
+          ]}
+        />,
+      );
+
+      expect(getByText("3 selected")).toBeInTheDocument();
+    });
+
+    it("should call onClearAll when the 'Clear' button is clicked", () => {
+      const { getByText } = render(
+        <ComboboxContentHeader
+          onSelectAll={onSelectAll}
+          onClearAll={onClearAll}
+          selectedCount={4}
+          searchValue=""
+          options={[
+            { id: "1", label: "Bilbo Baggins" },
+            { id: "2", label: "Frodo Baggins" },
+            { id: "3", label: "Samwise Gamgee" },
+            { id: "4", label: "Meriadoc Brandybuck" },
+          ]}
+        />,
+      );
+
+      fireEvent.click(getByText("Clear"));
+
+      expect(onClearAll).toHaveBeenCalledTimes(1);
+    });
+
+    describe("When the search term matches one or more options", () => {
+      it("should render a button that is labelled 'Clear'", () => {
+        const { getByText } = render(
+          <ComboboxContentHeader
+            onSelectAll={onSelectAll}
+            onClearAll={onClearAll}
+            selectedCount={1}
+            searchValue="Bag"
+            options={[
+              { id: "1", label: "Bilbo Baggins" },
+              { id: "2", label: "Frodo Baggins" },
+              { id: "3", label: "Samwise Gamgee" },
+            ]}
+          />,
+        );
+
+        expect(getByText("Clear")).toBeInTheDocument();
+      });
+    });
+
+    describe("When the search term does not match any options", () => {
```

Comment:
same thing here, on line `94` we have `When the search term matches one or more options`

let's group these up!